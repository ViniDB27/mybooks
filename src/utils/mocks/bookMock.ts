export const bookMock = {
    kind: "Mock",
    id: "1",
    etag: "Mock",
    selfLink: "Mock",
    volumeInfo: {
      title: "Mock",
      authors: ["Mock"],
      publisher: "Mock",
      publishedDate: "Mock",
      description: "Mock",
      pageCount: 10,
      imageLinks: {
        smallThumbnail: "Mock",
        thumbnail: "Mock",
      },
      language: "Mock",
    },
  };