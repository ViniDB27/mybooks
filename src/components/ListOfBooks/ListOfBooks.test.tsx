import render, { ReactTestRendererJSON } from "react-test-renderer";

import { ListOfBooks } from "./index";

import { bookMock } from "../../utils/mocks/bookMock";

describe("<HeaderWithTextInput />", () => {
  const tree = render
    .create(
      <ListOfBooks listBooks={[bookMock, bookMock]} onSlectedBook={() => {}} />
    )
    .toJSON() as ReactTestRendererJSON;

  it("verify if component is not null ", () => {
    expect(tree).not.toEqual(null);
  });

  it("verify if have 1 child", () => {
    expect(tree.children?.length).toEqual(1);
  });

  it("verify if child is a RCTScrollView", () => {
    const scrolView = (tree.children as any[])[0];
    expect(scrolView.type).toEqual("RCTScrollView");
  });

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("verify if type is View", () => {
    expect(tree.type).toEqual("View");
  });
});
