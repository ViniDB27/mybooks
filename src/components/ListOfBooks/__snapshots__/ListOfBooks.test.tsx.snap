// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderWithTextInput /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "marginLeft": 25,
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    <View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
                "flexDirection": "row",
                "height": 85,
                "marginVertical": 10,
                "overflow": "hidden",
                "padding": 5,
                "width": "85%",
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": 5,
                  "width": 60,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "Mock",
                  }
                }
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                  }
                }
              >
                Mock
              </Text>
              <Text>
                Mock...
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
                "flexDirection": "row",
                "height": 85,
                "marginVertical": 10,
                "overflow": "hidden",
                "padding": 5,
                "width": "85%",
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": 5,
                  "width": 60,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "Mock",
                  }
                }
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                  }
                }
              >
                Mock
              </Text>
              <Text>
                Mock...
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
